src/native/managed/cdacreader/Microsoft.Diagnostics.DataContractReader.Contracts/Contracts/StackWalk/Context/ContextHolder.cs

```diff
@@ -0,0 +1,48 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Diagnostics.DataContractReader.Contracts.StackWalkHelpers;
+
+public class ContnextHolder<T> : IPlatformAgnosticContext where T : struct, IPlatformContext
+{
+    public T Context;
+
+    public uint Size => Context.Size;
+    public uint DefaultContextFlags => Context.DefaultContextFlags;
+
+    public TargetPointer StackPointer { get => Context.StackPointer; set => Context.StackPointer = value; }
+    public TargetPointer InstructionPointer { get => Context.InstructionPointer; set => Context.InstructionPointer = value; }
+    public TargetPointer FramePointer { get => Context.FramePointer; set => Context.FramePointer = value; }
+
+    public unsafe void ReadFromAddress(Target target, TargetPointer address)
+    {
+        Span<byte> buffer = new byte[Size];
+        target.ReadBuffer(address, buffer);
+        FillFromBuffer(buffer);
+    }
+    public unsafe void FillFromBuffer(Span<byte> buffer)
+    {
+        Span<T> structSpan = MemoryMarshal.CreateSpan(ref Context, 1);
+        Span<byte> byteSpan = MemoryMarshal.Cast<T, byte>(structSpan);
+        if (buffer.Length > sizeof(T))
+        {
+            buffer.Slice(0, sizeof(T)).CopyTo(byteSpan);
+        }
+        else
+        {
+            buffer.CopyTo(byteSpan);
+        }
+    }
+    public unsafe byte[] GetBytes()
+    {
+        Span<T> structSpan = MemoryMarshal.CreateSpan(ref Context, 1);
```

Comment:
This should just be new Span<T>(ref Context)  .... I also think you should probably make T have the unmanaged constraint.